packages/payment/src/services/payment-module.ts

```diff
@@ -206,23 +207,38 @@ export default class PaymentModuleService<
     data: CreatePaymentSessionDTO,
     @MedusaContext() sharedContext?: Context
   ): Promise<PaymentSessionDTO> {
-    const sessionData = await this.paymentProviderService_.createSession(
-      data.provider_id,
-      data.providerContext
-    )
-
     const created = await this.paymentSessionService_.create(
       {
         provider_id: data.provider_id,
         amount: data.providerContext.amount,
         currency_code: data.providerContext.currency_code,
         payment_collection: paymentCollectionId,
-        data: sessionData,
       },
       sharedContext
     )
 
-    return await this.baseRepository_.serialize(created, { populate: true })
+    try {
+      const sessionData = await this.paymentProviderService_.createSession(
```

Comment:
suggestion: communicating with an external third party API shouldn't happen within a transaction. We should probably split that in a workflow. should we revisit that ? eventually, for now it could be moved one level higher and then call a protected `createPaymentSession_` which would be wrapped in a transaction

Reply:
Yes, this will be revisited as we introduce workflows.